cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -313,27 +316,32 @@ public void testVersionedProgramStartStopStatus() throws Exception {
     Assert.assertEquals(STOPPED, getProgramStatus(wordFrequencyService1));
     // start service
     startProgram(wordFrequencyService1, 200);
-    waitState(wordFrequencyService1, RUNNING);
+    verifyProgramRuns(wordFrequencyService1, ProgramRunStatus.RUNNING);
     // wordFrequencyService2 is stopped initially
     Assert.assertEquals(STOPPED, getProgramStatus(wordFrequencyService2));
     // start service in version2
     startProgram(wordFrequencyService2, 200);
-    waitState(wordFrequencyService2, RUNNING);
+    verifyProgramRuns(wordFrequencyService2, ProgramRunStatus.RUNNING);
     // wordFrequencyServiceDefault is stopped initially
     Assert.assertEquals(STOPPED, getProgramStatus(wordFrequencyServiceDefault));
     // start service in default version
     startProgram(wordFrequencyServiceDefault, 200);
-    waitState(wordFrequencyServiceDefault, RUNNING);
+    verifyProgramRuns(wordFrequencyServiceDefault, ProgramRunStatus.RUNNING);
     // same service cannot be run concurrently in the same app version
     startProgram(wordFrequencyService1, 409);
     stopProgram(wordFrequencyService1, null, 200, null);
+    verifyProgramRuns(wordFrequencyService1, ProgramRunStatus.KILLED);
     Assert.assertEquals(STOPPED, getProgramStatus(wordFrequencyService1));
     // wordFrequencyService1 can be run after wordFrequencyService1 is stopped
     startProgram(wordFrequencyService1, 200);
+    verifyProgramRuns(wordFrequencyService1, ProgramRunStatus.RUNNING);
 
     stopProgram(wordFrequencyService1, null, 200, null);
+    verifyProgramRuns(wordFrequencyService1, ProgramRunStatus.KILLED);
```

Comment:
move this and the following two waits together so that the total wait time can be reduced. you can move the waits together if the following waits don't depend on the current wait to finish.